### PR TITLE
Capture Firecracker process resources usage during sandbox run

### PR DESCRIPTION
**Pull request by 0div**
# Description
[E2B-944](https://linear.app/e2b/issue/E2B-944/capture-firecracker-process-resources-usage-during-sandbox-run)
- Get PID of `unshare` command after start
- Create `Sandbox.Stats()` method to return CPU% and MemInfo of the above PPID's children leveraging [gopsutil](https://github.com/shirou/gopsutil/)
- Create a `make stress` entry to specifically run stress tests.
- Created utilities to SSH into the sandbox, install and run [sysbench](https://github.com/akopytov/sysbench)
- Cleanup potentially dangling firecrackers processes at the end of test

# Test 
```sh
cd packages/orchestrator
make stress
```

### CODE REVIEW

**Comment by ValentaTomas** [comment]
Quick notes:
- Does capturing the `unshare` CPU (and RAM) work for getting the Firecracker resources?
- Will you be adding the hugepages RAM consumption or it is captured too already?

File: packages/orchestrator/internal/sandbox/sandbox.go [hunk 1]
```diff
@@ -443,3 +444,41 @@ func (s *Sandbox) SlotIdx() int {
...
+
+type SandboxStats struct {
+	CPUPercent float64 `json:"cpu_percent"`
+	MemoryVMS  uint64  `json:"memory_vms"`
+	MemoryRSS  uint64  `json:"memory_rss"`
+}
+
+func (s *Sandbox) Stats() (SandboxStats, error) {
+	stats := SandboxStats{
+		CPUPercent: 0,
```
Comment (ValentaTomas) [inline packages/orchestrator/internal/sandbox/sandbox.go]: Are you explicitly initializing the struct fields for clarity?

Comment (0div) [reply packages/orchestrator/internal/sandbox/sandbox.go]: It's that Go extension that kept bugging me about initializing the values, I don't usually do it.

**Comment by 0div** [comment]
> - Does capturing the `unshare` CPU (and RAM) work for getting the Firecracker resources?

I will write a more comprehensive test to make sure of that. 

> - Will you be adding the hugepages RAM consumption or it is captured too already?

My goal was to have this without adding the untouched hugepage to the total, does that make sense?

**Comment by 0div** [comment]
Wrote a test that SSHs into the sanbox, installs sysbench and runs as follows:
```
[12:37:02] [PID: 1731766] [Name: firecracker] CPU (%): 168.16 | VMS (MB): 521.25 | RSS (MB): 1.50 | Swap (MB): 0.00 | Stack: 0.00
[12:37:04] [PID: 1731766] [Name: firecracker] CPU (%): 186.73 | VMS (MB): 521.25 | RSS (MB): 1.50 | Swap (MB): 0.00 | Stack: 0.00
[12:37:06] [PID: 1731766] [Name: firecracker] CPU (%): 193.07 | VMS (MB): 521.25 | RSS (MB): 1.50 | Swap (MB): 0.00 | Stack: 0.00
[12:37:08] [PID: 1731766] [Name: firecracker] CPU (%): 196.82 | VMS (MB): 521.25 | RSS (MB): 1.50 | Swap (MB): 0.00 | Stack: 0.00
Stressing instance test-instance-1
$ sysbench memory --memory-block-size=128M --memory-total-size=1600M --memory-oper=write --verbosity=0 run
[12:37:10] [PID: 1731766] [Name: firecracker] CPU (%): 198.66 | VMS (MB): 521.25 | RSS (MB): 1.50 | Swap (MB): 0.00 | Stack: 0.00
[12:37:13] [PID: 1731766] [Name: firecracker] CPU (%): 199.12 | VMS (MB): 521.25 | RSS (MB): 1.50 | Swap (MB): 0.00 | Stack: 0.00
[12:37:15] [PID: 1731766] [Name: firecracker] CPU (%): 199.52 | VMS (MB): 521.25 | RSS (MB): 1.50 | Swap (MB): 0.00 | Stack: 0.00
[12:37:17] [PID: 1731766] [Name: firecracker] CPU (%): 199.73 | VMS (MB): 521.25 | RSS (MB): 1.50 | Swap (MB): 0.00 | Stack: 0.00
[12:37:19] [PID: 1731766] [Name: firecracker] CPU (%): 200.01 | VMS (MB): 521.25 | RSS (MB): 1.50 | Swap (MB): 0.00 | Stack: 0.00
[...]
[12:39:06] [PID: 1731766] [Name: firecracker] CPU (%): 202.01 | VMS (MB): 521.25 | RSS (MB): 1.50 | Swap (MB): 0.00 | Stack: 0.00
```
I'm not seeing any mem change for now, will have to look into it further into this.

**Pull request closed without merging**